assets/src/components/activities/file_upload/utils.ts

```diff
@@ -0,0 +1,50 @@
+import { DEFAULT_PART_ID } from 'components/activities/common/utils';
+import { FileUploadSchema, FileSpec } from 'components/activities/file_upload/schema';
+import { Responses } from 'data/activities/model/responses';
+import { GradingApproach, makeHint, makeStem, ScoringStrategy } from '../types';
+
+export const defaultModel: () => FileUploadSchema = () => {
+  return {
+    stem: makeStem(''),
+    fileSpec: createDefaultFileSpec(),
+    authoring: {
+      parts: [
+        {
+          id: DEFAULT_PART_ID,
+          scoringStrategy: ScoringStrategy.average,
+          gradingApproach: GradingApproach.manual,
+          responses: Responses.forTextInput(),
+          hints: [makeHint(''), makeHint(''), makeHint('')],
+        },
+      ],
+      transformations: [],
+      previewText: '',
+    },
+  };
+};
+
+export const createDefaultFileSpec: () => FileSpec = () =>
+  ({
+    maxCount: 1,
+    accept: '',
+  } as FileSpec);
+
+export function fileName(url: string) {
+  return url.substring(url.lastIndexOf('/') + 1);
+}
+
+export function toDate(ms: number) {
+  const date = new Date(ms);
+  return date.toLocaleString();
+}
+
+export function getReadableFileSizeString(fileSizeInBytes: number) {
+  let i = -1;
+  const byteUnits = [' kB', ' MB', ' GB', ' TB', 'PB', 'EB', 'ZB', 'YB'];
```

Comment:
lol, well I feel like > TB is an unlikely scenario :D, actually I hope we don't even reach GB honestly

Reply:
Yes indeed.  I "cloned and owned" this impl from somewhere, and simply left it to be quite scalable! lol